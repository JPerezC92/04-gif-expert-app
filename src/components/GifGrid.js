import GifGridItem from "./GifGridItem";
import useFetchGifs from "../hoooks/useFetchGifs";

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && (
        <p className="animate__animated animate__flash animate__infinite">
          cargando...
        </p>
      )}

      <div className="card-grid">
        {images.map((img) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

export default GifGrid;
