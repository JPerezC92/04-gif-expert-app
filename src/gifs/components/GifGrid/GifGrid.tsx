import { Box } from '@mui/material';
import React from 'react';

import { Gif } from '@/gifs/components/Gif';
import { type Gif as GifModel } from '@/gifs/domain';

import styles from './GifGrid.module.css';

interface Props {
	gifList: GifModel[];
}

export function GifGrid({ gifList }: Props) {
	return (
		<Box
			component='ol'
			paddingInlineStart={0}
			margin={0}
			marginInline='2px'
			marginBlockStart={2}
			display='grid'
			gridTemplateColumns='repeat(auto-fit, minmax(min(15rem,100%),1fr))'
			gridAutoRows='250px'
			gridAutoFlow='row dense'
			gridTemplateRows='masonry'
		>
			{gifList?.map(gif => (
				<Box
					key={gif.id}
					component='li'
					sx={{ listStyle: 'none' }}
					className={styles.Gif}
				>
					<Gif gif={gif} />
				</Box>
			))}
		</Box>
	);
}
