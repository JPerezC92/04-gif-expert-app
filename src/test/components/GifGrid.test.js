import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import useFetchGifs from "../../hooks/useFetchGifs";
jest.mock("../../hooks/useFetchGifs");

describe("Pruebas en el componente <GifGrid />", () => {
  const category = "cat";

  test("Debe de compararse correctamente con la imagen", () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar items cuando se cargan las imagenes useFetchGifs", () => {
    const gifs = [
      {
        id: "ABCD",
        title: "awesome funny gif",
        url: "https://localhost/any/thing.gif",
      },
      {
        id: "BFSFDS",
        title: "2 awesome funny gif",
        url: "https://localhost/any/thing.gif",
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    // expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length);
  });
});
