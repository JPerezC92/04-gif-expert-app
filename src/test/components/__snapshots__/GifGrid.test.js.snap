// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de compararse correctamente con la imagen 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    cat
  </h3>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
